File: src/main/java/ro/esolutions/eipl/models/EmployeeEquipmentModel.java
package ro.esolutions.eipl.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotNull;
import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class EmployeeEquipmentModel {

    private Long id;

    @NotNull
    private EmployeeModel employee;

    @NotNull
    private EquipmentModel equipment;

    @NotNull
    private LocalDate startDate;

    @NotNull
    private LocalDate endDate;
}

File: webapp/src/app/employee-equipment/employee-equipment.model.ts
import {Employee} from "../employees/employee.model";
import {EquipmentModel} from "../equipments/model/equipment.model";

export class EmployeeEquipmentModel {
  constructor(public id: number,
              public employee: Employee,
              public equipment: EquipmentModel,
              public startDate: Date,
              public endDate: Date
              ) {
  }
}
